### ISSUE TITLE
BCP data is wrong

### ISSUE THREAD

**Issue by DariukaB** [robbrad/UKBinCollectionData#661]
### Name of Council

BCP

### Issue Information

Bournemouth-Christchurch-Poole data is wrong
Eg. For BH1 3RZ collections for general waste and recycle are always on Tuesday and garden on Wednesday. Now it shows all on monday

### Verification

- [X] I searched for similar issues at https://github.com/robbrad/UKBinCollectionData/issues?q=is:issue and found no duplicates
- [X] I have checked my address/postcode/UPRN works on the council's website
- [X] I have provided a detailed explanation of the issue as well as steps to replicate the issue
- [X] I understand that this project is run by volunteer contributors therefore completion of this issue cannot be guaranteed

**Comment by dp247** [comment]
Guessing the API has messed up with daylight saving since everything is 11PM instead of midnight(?):
```json

{
    "BinType": "Food Waste",
    "PdfLink": "",
    "Communal": "false",
    "Next": "4/8/2024 11:00:00 PM",
    "Subsequent": "4/15/2024 11:00:00 PM"
  },
  
  
  
  
  
  {
    "BinType": "Garden Waste",
    "PdfLink": "https://www.bcpcouncil.gov.uk/gardenbincollectionsWed1",
    "Communal": "false",
    "Next": "4/9/2024 11:00:00 PM",
    "Subsequent": "4/23/2024 11:00:00 PM"
  },
  
  
  
  
  
  {
    "BinType": "Recycling",
    "PdfLink": "https://www.bcpcouncil.gov.uk/bincollectionbctue2",
    "Communal": "false",
    "Next": "4/15/2024 11:00:00 PM",
    "Subsequent": "4/29/2024 11:00:00 PM"
  },
  
  
  
  
  
  {
    "BinType": "Rubbish",
    "PdfLink": "",
    "Communal": "false",
    "Next": "4/8/2024 11:00:00 PM",
    "Subsequent": "4/22/2024 11:00:00 PM"
  }

```

**Comment by JherekC62** [comment]
Just to add a "me too@ for what it's worth, all the days are showing as a day early

**Comment by dp247** [comment]
> Just to add a "me too@ for what it's worth, all the days are showing as a day early
> 
> Is this likely an issue with BCP or with the integration itself? (in regards the DST issue)

I'll try localise the timezone, I think it's most likely the integration, but I'm too lazy to pick through the javascript they use to find out lol

**Comment by JherekC62** [comment]
I've just checked one of my bins history in HA. It actually started showing a day early on 29th March, which is before the change to DST? Recycling bin is collected on a Thursday, **not** Wednesday as it's showing now. The other bins are also showing the same, a day early. Hopefully this is useful information. 

**29 March 2024**
xxxxxxxx Road Recycling changed to Future: 2024-04-10
13:49:14 - 2 weeks ago
xxxxxxxx Road Recycling changed to Future: 2024-04-10
01:48:37 - 2 weeks ago

**28 March 2024**
xxxxxxxx Road Recycling changed to Today
01:48:37 - 2 weeks ago

**27 March 2024**
xxxxxxxx Road Recycling changed to Tomorrow
01:48:37 - 2 weeks ago

**25 March 2024**
xxxxxxxx Road Recycling changed to This Week: Thursday
01:48:33 - 2 weeks ago

**Comment by dp247** [comment]
> I've just checked one of my bins history in HA. It actually started showing as a day early on 29th March, which is before the change to DST? Recycling bin is collected on a Thursday, **not** Wednesday as it's showing now. The other bins are also showing the same, a day early. Hopefully this is useful information.
> 
> **29 March 2024** xxxxxxxx Road Recycling changed to Future: 2024-04-10 13:49:14 - 2 weeks ago xxxxxxxx Road Recycling changed to Future: 2024-04-10 01:48:37 - 2 weeks ago
> 
> **28 March 2024** xxxxxxxx Road Recycling changed to Today 01:48:37 - 2 weeks ago
> 
> **27 March 2024** xxxxxxxx Road Recycling changed to Tomorrow 01:48:37 - 2 weeks ago
> 
> **25 March 2024** xxxxxxxx Road Recycling changed to This Week: Thursday 01:48:33 - 2 weeks ago

Very helpful, thank you. Should be fixed with #673 - the 12-hour hour is parsed correctly now, then rounded to midnight - works on the test data at least 😄
